### PR TITLE
Dfr 2879 refactor pba validate controller to use handler architecture

### PR DESCRIPTION
**Pull request by hamzahtahirhmcts**
### Jira link (if applicable)

[DFR-2879](https://tools.hmcts.net/jira/browse/DFR-2879)

### Change description ###
Refactor PbaValidateController to use handler architecture

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        caseDetails.setData(caseData);
+
+        Mockito.when(pbaValidationService.isValidPBA(any(), eq("InvalidPbaNumber"))).thenReturn(false);
+
+        FinremCallbackRequest request = FinremCallbackRequest.builder()
+            .caseDetails(caseDetails)
+            .build();
+        var response = handler.handle(request, "userAuthorisation");
+
+        System.out.println("Response: " + response.getErrors());
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java:75]: Remove calls to System.out

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    @Test
+    void testHandle_InvalidPbaNumber_thenReturnError() {
+        FinremCaseDetails caseDetails = new FinremCaseDetails();
+        FinremCaseData caseData = new FinremCaseData();
+        caseData.setHelpWithFeesQuestion(NO);
+        caseData.setPbaNumber("InvalidPbaNumber");
+        caseDetails.setData(caseData);
+
+        Mockito.when(pbaValidationService.isValidPBA(any(), eq("InvalidPbaNumber"))).thenReturn(false);
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java:68]: `Mockito.when(pbaValidationService.isValidPBA("userAuthorisation", "InvalidPbaNumber")).thenReturn(false);`

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    @Test
+    void testHandle_ValidPbaNumber_thenHandlerCanHandle() {
+        FinremCaseDetails caseDetails = new FinremCaseDetails();
+        FinremCaseData caseData = new FinremCaseData();
+        caseData.setHelpWithFeesQuestion(NO);
+        caseData.setPbaNumber("ValidPbaNumber");
+        caseDetails.setData(caseData);
+
+        Mockito.when(pbaValidationService.isValidPBA(any(), eq("ValidPbaNumber"))).thenReturn(true);
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java:49]: `Mockito.when(pbaValidationService.isValidPBA("userAuthorisation", "InvalidPbaNumber")).thenReturn(false);`

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            .caseDetails(caseDetails)
+            .build();
+        var response = handler.handle(request, "userAuthorisation");
+
+        System.out.println("Response: " + response.getErrors());
+        System.out.println("Response: " + response);
+
+
+        assertThat(response).isNotNull();
+        assertThat(response.getErrors()).isNotEmpty();
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java:80]: I would make the test more precise. There should only be 1 error message:
`.size(), is(1));`
Similar thing with the following line. Use
`assertThat(response.getErrors.get(0)).isEqualTo(....`

**Review by jthmcts — CHANGES_REQUESTED** [review]
A few minor things to update but looks good otherwise.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java (hunk 1 above)
Comment (hamzahtahirhmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java]: [f9b4d98](https://github.com/hmcts/finrem-case-orchestration-service/pull/1610/commits/f9b4d98689cbcad345eb5410e9ce780382048443)

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java (hunk 4 above)
Comment (hamzahtahirhmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java]: [96b834e](https://github.com/hmcts/finrem-case-orchestration-service/pull/1610/commits/96b834ed0ba13abb53a65ceb5bb5610485f66cc0)

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java (hunk 3 above)
Comment (hamzahtahirhmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java]: [96b834e](https://github.com/hmcts/finrem-case-orchestration-service/pull/1610/commits/96b834ed0ba13abb53a65ceb5bb5610485f66cc0)

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java (hunk 2 above)
Comment (hamzahtahirhmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/casesubmission/CaseSubmissionPbaValidateMidEventHandlerTest.java]: [96b834e](https://github.com/hmcts/finrem-case-orchestration-service/pull/1610/commits/96b834ed0ba13abb53a65ceb5bb5610485f66cc0)

**Review by jthmcts — DISMISSED** [review]

**Review by jthmcts — APPROVED** [review]

**Review by so99y — APPROVED** [review]

**Pull request merged**